### PR TITLE
Wip php tag support

### PR DESCRIPTION
**Pull request by renz45**
I'm not actually going to merge this in at this point. I wanted to open a pr to make code review a bit easier.

### CODE REVIEW

File: src/html-grammar.pegjs [hunk 1]
```diff
@@ -548,6 +549,45 @@ text_node "Text Node"
   };
 }
 
+/* Inline php code */
+php "PHP Code"
+  = php_open com:(php_block) cc:(php_close)?
```
Comment (nwronski) [inline src/html-grammar.pegjs]: You can't have `php_close` be an optional item (e.g. `?`) if you want to match the end tag _or_ the end of the file.

File: src/html-grammar.pegjs [hunk 2]
```diff
@@ -548,6 +549,45 @@ text_node "Text Node"
...
+  / EOF
+  
+php_block
+  = s pb:php_scan s
+  {
+    var tn = pb !== null ? _u.textNode(pb) : '';
+    if(tn.indexOf('?>') !== -1) {
+      return error("Cannot have a php closing block " + esc.val('?>') + " inside of a PHP code block");
+    }
+    return {
```
Comment (nwronski) [inline src/html-grammar.pegjs]: Could we change the output format of the PHP tag so it ends up returning this format:

``` javascript
{
  "type": "php",
  "attributes": {},
  "contents":"var x = 23;"
}
```

Comment (renz45) [inline src/html-grammar.pegjs]: done and done

**Comment by nwronski** [comment]
This is good to go! 👍

**Pull request closed without merging**